libs/core/lci_base/src/lci_environment.cpp

```diff
@@ -233,6 +238,13 @@ namespace hpx { namespace util {
         }
     }
 
+    bool lci_environment::do_progress()
+    {
+        LCI_error_t ret = LCI_progress(LCI_UR_DEVICE);
+        HPX_ASSERT(ret == LCI_OK || ret == LCI_ERR_RETRY);
```

Comment:
Note that asserts are disabled (compiled out) in release mode. Should this throw an exception instead?

Reply:
I think it is fine. I don't want to add an additional branch here in release mode. Currently, LCI_progress will not return a value other than LCI_OK and LCI_ERR_RETRY in any case.